### ISSUE TITLE
cxflow split args are inconsistent with mainloop paradigm

### ISSUE THREAD

**Issue by gnovis** [Cognexa/cxflow#5]
Split args assume fixed number of streams (3) whilst mainloop can be freely configured.

**Comment by FloopCZ** [comment]
I actually believe that there are multiple problems with the current paradigm of `split`. Sometimes, `split` may need to connect to a database, balance the training examples, etc. How to pass all the necessary arguments? Currently, one would put database info and balancing setting to the `dataset:` part of the config, but the split ratio to function arguments. That does not seem to be right. And what if there is also e.g., a `fetch` and a `postprocess` function on the dataset? How to call those? I believe we need something more general.

Food for thought:
Maybe we could have `cxflow dataset` command, which can call arbitrary function on the dataset, e.g., `cxflow dataset split`. Furthermore, for the parameters I suggest to keep the current style of constructing the dataset, i.e., `cxflow dataset split ratio=1,2,3`. With this, one can have a `split:` section in the config with the default values and using this command, the defaults get overriden. The `split` function would accept no arguments and read everything from the config (it is also consistent with our `train_stream`, `predict_stream` etc. methods).

I suppose this will need further discussion with @petrbel and @blazekadam.

**Comment by petrbel** [comment]
I agree, the values we've passed so far are not sufficient for more general cases.

I attach examples which are more consistent with current cxflow CLI. It doesn't support invokation of arbitrary dataset method (which is an overkill for splitting, isn't it?), though.

```bash
cxflow split config.yaml                                    # normal
cxflow split config.yaml 'dataset.split.db.user=petr'   # rewrite arg
```

```yaml
dataset:  &dataset
  module: datasets.fuel.mnist_dataset
  class: MNISTDataset
  split:
    file: split.csv
    train: 9
    valid: 2
    test: 1
    db: {user: petr, password: pwd, host: ...}
```

**Comment by blazekadam** [comment]
Thanks for the comments. I have quite a lot of points to this discussion. We ll continue with this offline.

**Comment by petrbel** [comment]
Closed by #65
